infrastructure/lnbitsapi/src/shared/lnbitsapi-logger.ts

```diff
@@ -0,0 +1,91 @@
+import * as AppInsights from 'applicationinsights';
+import { TelemetryClient } from 'applicationinsights';
+import { SeverityLevel } from 'applicationinsights/out/Declarations/Contracts';
+import { Logger } from 'tslog';
+
+export enum LogLevel {
+  CRITICAL = 'Critical',
+  ERROR = 'Error',
+  WARN = 'Warn',
+  INFO = 'Info',
+  VERBOSE = 'Verbose',
+}
+
+export class LnbitsApiLogger {
+  private readonly context?: string;
+  private readonly logger: Logger<any>;
+
+  constructor(context?: { name: string } | string) {
+    this.context = typeof context === 'string' ? context : context?.name;
+    this.logger = new Logger({
+      name: this.context ?? '',
+      type: 'pretty',
+      hideLogPositionForProduction: true,
+    });
+  }
+
+  log(level: LogLevel, message: string, error?: unknown) {
+    switch (level) {
+      case LogLevel.CRITICAL:
+        this.critical(message, error);
+        break;
+
+      case LogLevel.ERROR:
+        this.error(message, error);
+        break;
+
+      case LogLevel.WARN:
+        this.warn(message, error);
+        break;
+
+      case LogLevel.INFO:
+        this.info(message, error);
+        break;
+
+      case LogLevel.VERBOSE:
+        this.verbose(message, error);
+        break;
+    }
+  }
+
+  critical(message: string, error?: unknown) {
+    //this.trace(SeverityLevel.Critical, message, error);
+    this.logger.error(this.format(message, error));
+  }
+
+  error(message: string, error?: unknown) {
+    this.trace(SeverityLevel.Error, message, error);
+    this.logger.error(this.format(message, error));
+  }
+
+  warn(message: string, error?: unknown) {
+    //this.trace(SeverityLevel.Warning, message, error);
+    this.logger.warn(this.format(message, error));
+  }
+
+  info(message: string, error?: unknown) {
+    //this.trace(SeverityLevel.Information, message, error);
+    this.logger.info(this.format(message, error));
+  }
+
+  verbose(message: string, error?: unknown) {
+    //this.trace(SeverityLevel.Verbose, message, error);
+    this.logger.trace(this.format(message, error));
+  }

```

Comment:
I wasn't sure if this would work on the VM Docker. I suspect that I still need some configuration data so that I can access "AppInsights".

I wanted to clarify this with you as soon as you have time to look at these things.